ui/src/app/applications/components/filter/filter.tsx

```diff
@@ -0,0 +1,114 @@
+import {Autocomplete, CheckboxOption, CheckboxRow, Tooltip} from 'argo-ux';
+import * as React from 'react';
+
+import './filter.scss';
+
+interface FilterMap {
+    [label: string]: boolean;
+}
+
+interface FilterProps {
+    selected: string[];
+    setSelected: (items: string[]) => void;
+    options?: CheckboxOption[];
+    label?: string;
+    field?: boolean;
+    error?: boolean;
+    loading?: boolean;
+}
+
+export const Filter = (props: FilterProps) => {
+    const init = {} as FilterMap;
+    props.selected.forEach(s => (init[s] = true));
+
+    const [values, setValues] = React.useState(init);
+    const [tags, setTags] = React.useState([]);
+    const [input, setInput] = React.useState('');
+    const [collapsed, setCollapsed] = React.useState(false);
+
+    React.useEffect(() => {
+        const map: string[] = Object.keys(values).filter(s => values[s]);
+        props.setSelected(map);
+        if (props.field) {
+            setTags(
+                Object.keys(values).map(v => {
+                    return {label: v} as CheckboxOption;
+                })
+            );
+        }
+    }, [values]);
+
+    return (
+        <div className='filter'>
+            <div className='filter__header'>
+                {props.label || 'FILTER'}
+                {(props.selected || []).length > 0 || (props.field && Object.keys(values).length > 0) ? (
+                    <div
+                        className='argo-button argo-button--base argo-button--sm'
+                        style={{marginLeft: 'auto'}}
+                        onClick={() => {
+                            setValues({} as FilterMap);
+                            setInput('');
+                        }}>
+                        <i className='fa fa-times-circle' /> CLEAR
+                    </div>
+                ) : (
+                    <i className={`fa fa-caret-${collapsed ? 'down' : 'up'} filter__collapse`} onClick={() => setCollapsed(!collapsed)} />
+                )}
+            </div>
+            {!collapsed &&
+                (props.loading ? (
+                    <FilterLoading />
+                ) : props.error ? (
+                    <FilterError />
+                ) : (
+                    <React.Fragment>
+                        {props.field && (
+                            <Autocomplete
```

Comment:
* Autocomplete lost the ability to select option using the keyboard:


https://user-images.githubusercontent.com/426437/124003658-1079d200-d98c-11eb-8eb5-b6ef3183659d.mov